collector/library/pod.js

```diff
@@ -224,15 +210,57 @@ class Pod {
      * 
      */
     async submitDataToLogstash({ data, batch, }) {
-        const body = JSON.stringify({ message: data.toString() });
+        const dataAsString = data.toString();
+        let lines;
+
+        if (this.remainder.length > 0) {
+            lines = (this.remainder + dataAsString).split('\n');
+            this.remainder = '';
+        } else {
+            lines = dataAsString.split('\n');
+        }
 
-        await fetch('http://logstash:5000/', {
-            method: 'post',
-            body,
-            headers: { 'Content-Type': 'application/json' },
-        });
+        for (let i = 0; i < lines.length; i++) {
+            const line = lines[i];
+            let body;
+
+            if (this.isJSON(trim(line))) {
+                // We enter this condition only once we managed to get a one line JSON log
+                // Enrich the object
+                const dataObject = JSON.parse(trim(line));
+                dataObject._collector_batchId = batch.id;
+                dataObject._collector_source_url = this.targetUrl;
+                body = JSON.stringify(dataObject);
+            } else {
+                // From here we can have one of 2 options:
+                // 1. a partial JSON log line
+                // 2. a plain text log
+
+                if (i === lines.length - 1) { // This is the last line from []lines
+                    // This is a partial json, carry this remainder over to the next iteration of submitDataToLogstash()
```

Comment:
not accurate comment - even if it's partial it could still be text. the code would work just the same